doc/release/1.17.0-notes.rst

```diff
@@ -129,6 +129,15 @@ the stream changes for any given seed is extremely small. If a 0 is encountered
 underlying generator, then the incorrect value produced (either ``np.inf``
 or ``np.nan``) is now dropped.
 
+``arr.writeable`` can be set to true slightly more liberal
+----------------------------------------------------------
+
+In rare cases, it can now be possible to switch an array from not writeable
```

Comment:
I'd turn this around and state the problem first, like
"In rare cases, it was not possible to switch an array from not writeable 
to writeable, even if the base was writeable, because the search for the
base array had a bug. This is now fixed."

Or is this not true?

Reply:
Half true, I made it now:
```
In rare cases, it was not possible to switch an array from not writeable
to writeable, although a base array is writeable. This can happen if an
intermediate ``arr.base`` object is writeable. Previously, only the deepest
base object was considered for this decision. However, in rare cases this
object does not have the necessary information. In that case switching to
writeable was never allowed. This has now been fixed.
```
(will push in a minute)

Reply:
I'm still slightly confused, but I think would get the point that what was a problem is now fixed!

Reply:
It is a bit confusing. Basically, we do `arr.base.base.base` (until reaching the end, or a non-array). If the last one is not an array, "readonly" logic only works for buffers, not for `__array_interface__` or so. The change, is to check `writeable` on every base and stop early. Which fixes the case when the last base object is an object where we cannot find out whether or not it is writeable.